src/stores/appcatalog/actions.ts

```diff
@@ -1,22 +1,55 @@
 import GiantSwarm from 'giantswarm';
+import { IOAuth2User } from 'lib/OAuth2/OAuth2User';
+import CPClient from 'model/clients/CPClient';
+import { getAppCatalogs } from 'model/services/controlplane/appcatalogs/appcatalogs';
+import { IAppCatalog } from 'model/services/controlplane/appcatalogs/types';
+import { IState } from 'reducers/types';
+import FeatureFlags from 'shared/FeatureFlags';
+import { IAppCatalogsState, IStoredAppCatalog } from 'stores/appcatalog/types';
+import { getCPAuthUser } from 'stores/cpauth/selectors';
 
 import { createAsynchronousAction } from '../asynchronousAction';
 
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-export const listCatalogs = createAsynchronousAction<any, any, any>({
+export const listCatalogs = createAsynchronousAction<
+  undefined,
+  IState,
+  IAppCatalogsState
+>({
   actionTypePrefix: 'LIST_CATALOGS',
 
-  perform: async (_state) => {
-    const appsApi = new GiantSwarm.AppsApi();
-    const catalogs = await appsApi.getAppCatalogs(); // Use model layer?
+  perform: async (currentState: IState): Promise<IAppCatalogsState> => {
+    let catalogs: IAppCatalog[] = [];
+
+    let cpAuthUser: IOAuth2User | null = null;
+    if (FeatureFlags.FEATURE_CP_ACCESS) {
+      cpAuthUser = getCPAuthUser(currentState);
+    }
+
+    if (cpAuthUser) {
+      const client = new CPClient(
+        cpAuthUser.idToken,
+        cpAuthUser.authorizationType
+      );
+
+      catalogs = await getAppCatalogs(client);
```

Comment:
Using the CP API to fetch app catalogs, if the feature is supported and if the user is logged in.